gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/InsertAllRequest.java

```diff
@@ -0,0 +1,400 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Google Cloud BigQuery insert all request. This class can be used to stream data into BigQuery one
+ * record at a time without needing to run a load job. This approach enables querying data without
+ * the delay of running a load job. There are several important trade-offs to consider before
+ * choosing an approach.
+ *
+ * @see <a href="https://cloud.google.com/bigquery/streaming-data-into-bigquery">Streaming Data into
+ *     BigQuery</a>
+ */
+public class InsertAllRequest implements Serializable {
+
+  private static final long serialVersionUID = 211200307773853078L;
+
+  private final TableId table;
+  private final List<RowToInsert> rows;
+  private final Boolean skipInvalidRows;
+  private final Boolean ignoreUnknownValues;
+
+  /**
+   * A Google Big Query row to be inserted into a table. Each {@code RowToInsert} has an associated
+   * id used by BigQuery to detect duplicate insertion requests on a best-effort basis.
+   *
+   * <p>
+   * Example usage of creating a row to insert:
+   * <pre>    {@code
+   *   List<Long> repeatedFieldValue = Arrays.asList(1L, 2L);
+   *   Map<String, Object> recordContent = new HashMap<String, Object>();
+   *   recordContent.put("subfieldName1", "value");
+   *   recordContent.put("subfieldName2", repeatedFieldValue);
+   *   Map<String, Object> rowContent = new HashMap<String, Object>();
+   *   rowContent.put("fieldName1", true);
+   *   rowContent.put("fieldName2", recordContent);
+   *   RowToInsert row = new RowToInsert("rowId", rowContent);
+   * }</pre>
+   *
+   * @see <a href ="https://cloud.google.com/bigquery/streaming-data-into-bigquery#dataconsistency">
+   *     Data Consistency</a>
+   */
+  public static class RowToInsert implements Serializable {
+
+    private static final long serialVersionUID = 8563060538219179157L;
+
+    private final String id;
+    private final Map<String, Object> content;
+
+    RowToInsert(Map<String, Object> content) {
+      this.id = null;
+      this.content = checkNotNull(content);
+    }
+
+    /**
+     * Creates a row to be inserted.
+     *
+     * @param id id of the row, used to identify duplicates
+     * @param content the actual content of the row
+     */
+    public RowToInsert(String id, Map<String, Object> content) {
```

Comment:
Based off previous code, it looks like `gcloud-java` has the precedent of using `of` in this type of situation rather than expose the constructor.